mindmeld/tokenizer.py

```diff
@@ -45,65 +48,16 @@ def _init_regex(self):
         """
         # List of regex's for matching and tokenizing when keep_special_chars=False
         regex_list = []
-        # List of regex's for matching and tokenizing when keep_special_chars=True
-        keep_special_regex_list = []
 
-        exception_chars = "\@\[\]\|\{\}'"  # noqa: W605
-
-        # This is more of a tactical fix at this moment. Will probably need a more generic way
-        # to handle all currency symbols.
-        currency_symbols = "$¥"
-        to_exclude = currency_symbols + "".join(self.exclude_from_norm)
+        # fetches all currency symbols in unicode by iterating through the character set and
+        # selecting the currency symbols based on the unicode currency category 'Sc'
+        currency_symbols = u"".join(
+            chr(i) for i in range(0xFFFF) if unicodedata.category(chr(i)) == "Sc"
+        )
 
         letter_pattern_str = "[^\W\d_]+"  # noqa: W605
 
-        # Make keep special regex list
-        keep_special_regex_list.append(
-            "?P<start>^[^\w\d&" + to_exclude + exception_chars + "]+"
-        )  # noqa: W605
-        keep_special_regex_list.append(
-            "?P<end>[^\w\d&" + to_exclude + exception_chars + "]+$"
-        )  # noqa: W605
-        keep_special_regex_list.append(
-            "?P<pattern1>(?P<pattern1_replace>"
-            + letter_pattern_str  # noqa: W605
-            + ")"
-            + "[^\w\d\s&"
-            + exception_chars
-            + "]+(?=[\d]+)"
-        )  # noqa: W605
-        keep_special_regex_list.append(
-            "?P<pattern2>(?P<pattern2_replace>[\d]+)[^\w\d\s&"
-            + exception_chars  # noqa: W605
-            + "]+"
-            + "u(?="
-            + letter_pattern_str
-            + ")"
-        )  # noqa: W605
-        keep_special_regex_list.append(
-            "?P<pattern3>(?P<pattern3_replace>"
-            + letter_pattern_str
-            + ")"  # noqa: W605
-            + "[^\w\d\s&"
-            + exception_chars
-            + "]+"
-            + "(?="  # noqa: W605
-            + letter_pattern_str
-            + ")"
-        )  # noqa: W605
-        keep_special_regex_list.append(
-            "?P<escape1>(?P<escape1_replace>[\w\d]+)"
-            + "[^\w\d\s"  # noqa: W605
-            + exception_chars
-            + "]+"
-            + "(?=\|)"
-        )  # noqa: W605
-        keep_special_regex_list.append(
-            "?P<escape2>(?P<escape2_replace>[\]\}]+)"
-            + "[^\w\d\s"  # noqa: W605
-            + exception_chars
-            + "]+(?=s)"
-        )  # noqa: W605
+        to_exclude = currency_symbols + "".join(self.exclude_from_norm)
```

Comment:
if `exclude_from_norm` is None, won't this cause an error here?

Reply:
It's set as: ``self.exclude_from_norm = exclude_from_norm or []``